services/src/contexts/migrations/migration_0000_initial.rs

```diff
@@ -0,0 +1,105 @@
+use async_trait::async_trait;
+use tokio_postgres::Transaction;
+
+use crate::{
+    error::Result,
+    layers::{
+        add_from_directory::UNSORTED_COLLECTION_ID, storage::INTERNAL_LAYER_DB_ROOT_COLLECTION_ID,
+    },
+};
+
+use super::database_migration::{DatabaseVersion, Migration};
+
+pub struct Migration0000Initial;
+
+#[async_trait]
+impl Migration for Migration0000Initial {
+    fn prev_version(&self) -> Option<DatabaseVersion> {
+        None
+    }
+
+    fn version(&self) -> DatabaseVersion {
+        "0000_initial".into()
+    }
+
+    async fn migrate(&self, tx: &Transaction<'_>) -> Result<()> {
+        let config = crate::util::config::get_config_element::<crate::util::config::Postgres>()?;
+
+        let schema_name = &config.schema;
+
+        if schema_name != "pg_temp" {
+            tx.batch_execute(&format!("CREATE SCHEMA IF NOT EXISTS {schema_name};",))
+                .await?;
+        }
+
+        tx.batch_execute(include_str!("migration_0000_initial.sql"))
+            .await?;
+
+        let stmt = tx
+            .prepare(
+                "
+            INSERT INTO geoengine (clear_database_on_start, database_version) VALUES ($1, '0000_initial');",
+            )
+            .await?;
+
+        tx.execute(&stmt, &[&config.clear_database_on_start])
+            .await?;
+
+        let stmt = tx
+            .prepare(
+                r#"
+            INSERT INTO layer_collections (
+                id,
+                name,
+                description,
+                properties
+            ) VALUES (
+                $1,
+                'Layers',
+                'All available Geo Engine layers',
+                ARRAY[]::"PropertyType"[]
+            );"#,
+            )
+            .await?;
+
+        tx.execute(&stmt, &[&INTERNAL_LAYER_DB_ROOT_COLLECTION_ID])
```

Comment:
This is critical. If we change the constant (and we will), this changes without being a new migration.

Reply:
what would be the solution? Create a new constant in the migration?

Reply:
I think we have to do this. You also could not say `UPDATE id = 1234 where ID = ?` and then add the constant since it could be changed and the update would not have an effect.

Reply:
045393894029fd54081e75acd1ba1c940f6bd3b8